diff --git a/node_modules/payload/dist/collections/init.js b/node_modules/payload/dist/collections/init.js
index 9383d60..2be62aa 100644
--- a/node_modules/payload/dist/collections/init.js
+++ b/node_modules/payload/dist/collections/init.js
@@ -62,10 +62,10 @@ function registerCollections(ctx) {
             });
             versionSchema.plugin(mongoose_paginate_v2_1.default, { useEstimatedCount: true })
                 .plugin(buildQuery_1.default);
-            ctx.versions[collection.slug] = mongoose_1.default.model(versionModelName, versionSchema);
+            ctx.versions[collection.slug] = mongoose_1.default.models[versionModelName] || mongoose_1.default.model(versionModelName, versionSchema);
         }
         ctx.collections[formattedCollection.slug] = {
-            Model: mongoose_1.default.model(formattedCollection.slug, schema),
+            Model: mongoose_1.default.models[formattedCollection.slug] || mongoose_1.default.model(formattedCollection.slug, schema),
             config: formattedCollection,
         };
         // If not local, open routes
diff --git a/node_modules/payload/dist/init.js b/node_modules/payload/dist/init.js
index f47f493..bba63f3 100644
--- a/node_modules/payload/dist/init.js
+++ b/node_modules/payload/dist/init.js
@@ -45,7 +45,11 @@ const init = (payload, options) => {
         .digest('hex')
         .slice(0, 32);
     payload.local = options.local;
-    payload.config = (0, load_1.default)(payload.logger);
+    if (options.validatedConfig) {
+        payload.config = options.validatedConfig
+    } else {
+        payload.config = (0, load_1.default)(payload.logger);
+    }
     // If not initializing locally, scaffold router
     if (!payload.local) {
         payload.router = express_1.default.Router();
